### PR TITLE
VSCode: Various code completion enhancements.

### PR DESCRIPTION
**Pull request by dbalek**
Couple of enhancements to VSCode code completion:
- Parameter names in lambda expressions should be based on type instead of name of the corresponding type parameters
- Hovers for local variables and parameter should provide basic info on var declaration
- Links in code completion documentation and hovers should be resolved to web URLs

### CODE REVIEW

**Comment by mbien** [comment]
this needs the https://github.com/apache/netbeans/labels/Java label since the changes are in java modules.

**Review by MartinBalin — APPROVED** [review]

File: java/java.editor/src/org/netbeans/modules/editor/java/Utilities.java [hunk 1]
```diff
@@ -93,6 +94,7 @@
 public final class Utilities {
     
     private static final String ERROR = "<error>"; //NOI18N
+    private static final Pattern LINK_PATTERN = Pattern.compile("<a href='(\\*\\d+)'>(.*?)<\\/a>", Pattern.CASE_INSENSITIVE | Pattern.MULTILINE);
```
Comment (sdedic) [inline java/java.editor/src/org/netbeans/modules/editor/java/Utilities.java:97]: Nitpick: `(.*?) -> (.*)` (* includes no repetition at all)

**Review by sdedic — APPROVED** [review]

File: java/java.editor/src/org/netbeans/modules/editor/java/Utilities.java (hunk 1 above)
Comment (dbalek) [reply java/java.editor/src/org/netbeans/modules/editor/java/Utilities.java:97]: `.*`  is a greedy quantifier that matches as many characters as possible while `.*?` is a lazy quantifier that matches as few characters as possible.

Comment (sdedic) [reply java/java.editor/src/org/netbeans/modules/editor/java/Utilities.java:97]: eh, sorry ... forgot about greedy/lazy modifier. thanks.

**Review by jlahoda — APPROVED** [review]
Seems reasonable to me.

**Pull request merged**